### DISCUSSION TITLE
Prime React in Next.js

### DISCUSSION THREAD

**Discussion by kamel996** [primefaces/community#1934] [PrimeReact] [answered]
Can i use the PrimeReactProvider in next js main.tsx ? or it will make all the child components treated as client components ?

**Comment by Santiago-Pe** [comment, marked as answer]
No, on Next if you pass a server component as a chilldren to a client component, still beeing sever component, but only as a chidlren. Insted, if you import a server component on client component, this turn into client component

**Reply by Bader975** [reply]
so i don't have to use ``use client ``directive on layout.tsx ?
